backend-nannyway/src/main/java/de/neuefische/nannyway/service/ChildInCareService.java

```diff
@@ -39,11 +43,31 @@ public ChildInCare addChildInCare(String firstName, String lastName, LocalDate b
         return kidsDb.save(kid);
     }
 
-    public void deleteKidById(String id){
+    public void deleteKidById(String id) {
         kidsDb.deleteById(id);
     }
 
-    public List<ChildInCare> getChildByNanny (String nannyUsername){
+    public List<ChildInCare> getChildByNanny(String nannyUsername) {
         return kidsDb.findByNanny(nannyUsername);
     }
+
+    public ChildInCare getChildById(String id) {
+        Optional<ChildInCare> optionalChildInCare = kidsDb.findById(id);
+        if (optionalChildInCare.isPresent()) return optionalChildInCare.get();
+        throw new ResponseStatusException(HttpStatus.NOT_FOUND);
+    }
+
+    public ChildInCare updateKid(String id, String key, EditChildInCareDto editChildInCareDto) {
+        ChildInCare childInCareUpdate = getChildById(id);
+        switch (key) {
+            case "pickUpPerson" -> childInCareUpdate.setPickUpPerson(editChildInCareDto.getPickUpPerson());
+            case "infoText" -> childInCareUpdate.setInfoText(editChildInCareDto.getInfoText());
+            case "hoursInCarePerWeek" -> childInCareUpdate.setHoursInCarePerWeek(editChildInCareDto.getHoursInCarePerWeek());
+            case "contractTerm" -> childInCareUpdate.setContractTerm(editChildInCareDto.getContractTerm());
+            case "phoneNumber" -> childInCareUpdate.setPhoneNumber(editChildInCareDto.getPhoneNumber());
+            case "email" -> childInCareUpdate.setEmail(editChildInCareDto.getEmail());
+            default -> throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "updateKey not valid");
+        }
+        return kidsDb.save(childInCareUpdate);
+    }
```

Comment:
If your EditChildInCareDto had all the attributes of your ChildInCare model (I'm not sure, but maybe you don't even need the dto and can use the actual model), you could just send the whole childInCare to your backend and save it in the db. Since the objects id is unique(?) the update ChildInCare would overwrite the old version. This way you wouldn't need a key or have to loop through the model.

Reply:
I see your point and in fact tried that approach for approx. half of the day 😬  But I need the key to define different endpoints in the Controller class in order to be able to update specific database elements individually via separate single line forms. Therefore this dto and using the switch makes the most sense ... approved by Fabian himself. But thanks for really diving into my code. appreciate 😻